### DISCUSSION TITLE
use with fzf

### DISCUSSION THREAD

**Discussion by sweetbbak** [hpjansson/chafa#139] [Q&A] [answered]
has anyone been able to get Chafa to work with the preview function on fzf? Kitty's icat recently broke after an update. So far ueberzug is the only one that still works to this day, which is kind of wild because its archived and gone. It sucks because I was using this functionality for a ton of really cool scripts for managing libraries of music, books, manga etc... I was wondering if anyone has got Chafa to work with fzf or maybe someone can clue me in on the matter

**Comment by hpjansson** [comment, marked as answer]
Looks like someone made it work: https://old.reddit.com/r/commandline/comments/o7xkjp/fzf_preview_anything/

PeFClic's reply in that thread indicates it's possible, although it seems like fzf was unable to pass through the escape codes needed to make sixel or other more advanced image protocols work.

They seem to be using [Pistol](https://github.com/doronbehar/pistol) with a configuration file entry that looks like this:

`image/* chafa %pistol-filename%`

You may want to add `-f symbol` to the command line if it's printing garbage (sixels or kitty protocol being filtered by fzf). I haven't tested this myself, so YMMV :-)

**Comment by christosangelopoulos** [comment]
`chafa --format=symbols  -s 30x30 path/to/img.png`

`chafa  -s 30x30 path/to/img.png`  did not work in kitty, but adding `--format=symbols ` somehow did the trick.

**Reply by hpjansson** [reply]
Might be worth reporting an issue against fzf for this. I think it's mangling the escape codes somehow.

**Reply by christosangelopoulos** [reply]
It should be a fzf thing, I get similar trouble with img2sixel too.
